src/dtx/dtx_common.c

```diff
@@ -304,63 +329,91 @@ dtx_batched_commit(void *arg)
 				    struct dtx_batched_commit_args, dbca_link);
 		dtx_get_dbca(dbca);
 		cont = dbca->dbca_cont;
-
 		d_list_move_tail(&dbca->dbca_link, &dmi->dmi_dtx_batched_list);
 		dtx_stat(cont, &stat);
 
-		if ((stat.dtx_committable_count > DTX_THRESHOLD_COUNT) ||
-		    (stat.dtx_oldest_committable_time != 0 &&
-		     dtx_hlc_age2sec(stat.dtx_oldest_committable_time) >
-		     DTX_COMMIT_THRESHOLD_AGE)) {
+		if (!cont->sc_closing &&
+		    !dbca->dbca_deregister && dbca->dbca_commit_req == NULL &&
+		    ((stat.dtx_committable_count > DTX_THRESHOLD_COUNT) ||
+		     (stat.dtx_oldest_committable_time != 0 &&
+		      dtx_hlc_age2sec(stat.dtx_oldest_committable_time) >=
+		      DTX_COMMIT_THRESHOLD_AGE))) {
 			sleep_time = 0;
-			cnt = dtx_fetch_committable(cont, DTX_THRESHOLD_COUNT,
-						    NULL, DAOS_EPOCH_MAX,
-						    &dtes);
-			if (cnt > 0) {
-				rc = dtx_commit(cont, dtes, cnt, true);
-				dtx_free_committable(dtes, cnt);
-				if (rc != 0)
-					/* Not fatal, continue other batched
-					 * DTX commit, DTX aggregation.
-					 */
-					D_WARN("Fail to batched commit dtx: "
-					       DF_RC"\n", DP_RC(rc));
-
-				if (!cont->sc_dtx_aggregating ||
-				    !cont->sc_dtx_cleanup_stale)
-					dtx_stat(cont, &stat);
+			dtx_get_dbca(dbca);
+			dtx_init_sched_req(cont, &dbca->dbca_commit_req);
```

Comment:
You can't use sched_request like this, there are two ways to get a sched_request for target ULT:
1. Call sched_req_get(attr, NULL) from the target ULT;
2. Create target ULT first, then call sched_req_get(attr, target_thread) in different ULT;

BTW, it's better to make the ULT creation for these three ULTs into a wrapper function to avoid duplicated code.